drivers/SmartThings/matter-thermostat/src/test/test_matter_room_ac.lua

```diff
@@ -198,10 +199,16 @@ end
 test.register_coroutine_test(
   "Test profile change on init for Room AC device type",
   function()
+    mock_device_configure:set_field("__BATTERY_SUPPORT", "NO_BATTERY") -- since we're assuming this would have happened during device_added in this case.
     test.socket.device_lifecycle:__queue_receive({ mock_device_configure.id, "doConfigure" })
-    mock_device_configure:expect_metadata_update({ profile = "room-air-conditioner" })
     mock_device_configure:expect_metadata_update({ provisioning_state = "PROVISIONED" })
-
+    test.socket.matter:__queue_receive(
```

Comment:
```suggestion
    test.wait_for_events()
    test.socket.matter:__queue_receive(
```
I noticed this test case fails occasionally and just needs a `test.wait_for_events()` here to fix

Reply:
good catch!